sync-core/src/main/java/com/cloudant/sync/query/QueryValidator.java

```diff
@@ -146,6 +167,117 @@
         return accumulator;
     }
 
+    @SuppressWarnings("unchecked")
+    private static List<Object> compressMultipleNotOperators(List<Object> clause) {
+        List<Object> accumulator = new ArrayList<Object>();
+
+        for (Object fieldClause: clause) {
+            Object predicate;
+            String fieldName;
+            // if this isn't a map, we don't know what to do so add the clause
+            // to the accumulator to be dealt with later as part of the final selector
+            // validation.
+            if (fieldClause instanceof Map && !((Map) fieldClause).isEmpty()) {
+                Map<String, Object> fieldClauseMap = (Map<String, Object>) fieldClause;
+                fieldName = (String) fieldClauseMap.keySet().toArray()[0];
```

Comment:
I think in iOS we have the same symptom as in the Java implementation.  In iOS we have the `allKeys[0]` pattern in as many places as `keySet().toArray()[0]` in Java.  Your point is well taken but I think whatever we choose to do going forward should be applied to both implementations for consistency and like you said, it is late in the game to make a change.